sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -1870,4 +1870,19 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
       assert(sampled.count() == sampledOdd.count() + sampledEven.count())
     }
   }
+
+  test("SPARK-10707: nullability should be correctly propagated through set operations") {
+    withTempTable("src") {
+      Seq((1, 1)).toDF("key", "value").registerTempTable("src")
+      checkAnswer(
+        sql("""SELECT count(v) FROM (
+              |  SELECT v FROM (
+              |    SELECT 'foo' AS v FROM src UNION ALL
```

Comment:
spark SQL support `select xxx` without `from` clause, so we don't need to create the src table.